components/brave_wallet_ui/utils/format-prices.ts

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import BigNumber from 'bignumber.js'
+
+import { CurrencySymbols } from './currency-symbols'
+
+export const formatWithCommasAndDecimals = (value: string, decimals?: number) => {
+  // empty string indicates unknown balance
+  if (value === '') {
+    return ''
+  }
+
+  const valueBN = new BigNumber(value)
+
+  // We sometimes return Unlimited as a value
+  if (valueBN.isNaN()) {
+    return value
+  }
+
+  if (valueBN.isZero()) {
+    return '0.00'
+  }
+
+  const fmt = {
+    decimalSeparator: '.',
+    groupSeparator: ',',
+    groupSize: 3
+  } as BigNumber.Format
+
+  if (decimals && valueBN.isGreaterThan(10 ** -decimals)) {
+    return valueBN.toFormat(decimals, BigNumber.ROUND_UP, fmt)
+  }
+
+  if (valueBN.isGreaterThanOrEqualTo(10)) {
+    return valueBN.toFormat(2, BigNumber.ROUND_UP, fmt)
+  }
+
+  if (valueBN.isGreaterThanOrEqualTo(1)) {
+    return valueBN.toFormat(3, BigNumber.ROUND_UP, fmt)
+  }
+
+  return valueBN.toFormat(fmt)
+}
+
+export const formatFiatAmountWithCommasAndDecimals = (value: string, defaultCurrency: string): string => {
+  if (!value) {
+    return ''
+  }
+
+  const currencySymbol = CurrencySymbols[defaultCurrency]
+
+  // Check to make sure a formatted value is returned before showing the fiat symbol
+  if (!formatWithCommasAndDecimals(value, 2)) {
+    return ''
+  }
+  return currencySymbol + formatWithCommasAndDecimals(value, 2)
```

Comment:
we may want to handle local currencies where it's typical to display the symbol after the amount

Reply:
@diracdeltas I have changed the implementation to use `Intl.NumberFormat` which will handle placement of currency symbol in different locales as shown [here](https://github.com/brave/brave-core/blob/c9fb101fecbd25c015812ca0f5b25584890ae7bf/components/brave_wallet_ui/utils/amount.ts#L314)